TestModels/CallingAWSSDKFromLocalService/test/SimpleCallingAWSSDKFromLocalServiceImplTest.dfy

```diff
@@ -0,0 +1,92 @@
+
+  import opened SimpleCallingawssdkfromlocalserviceTypes
+  import opened Wrappers
+  method{:test} CallDDBScan(){
+    var client :- expect SimpleCallingAWSSDKFromLocalService.SimpleCallingAWSSDKFromLocalService();
+    TestCallDDBScan_Success(client);
+    TestCallDDBScan_Failure(client);
+  }
+
+  method TestCallDDBScan_Success(client: ISimpleCallingAWSSDKFromLocalServiceClient)
+      requires client.ValidState()
+      modifies client.Modifies
+      ensures client.ValidState()
+  {
+    var ddbClient :- expect Dynamodb.DynamoDBClient();
+    var resSuccess := client.CallDDBScan(SimpleCallingAWSSDKFromLocalService.Types.CallDDBScanInput(ddbClient := ddbClient, tableArn := TABLE_ARN_SUCCESS_CASE));
+
+    expect resSuccess.Success?;
+    expect resSuccess.value.itemOutput != -1;
+  }
+
+  method TestCallDDBScan_Failure(client: ISimpleCallingAWSSDKFromLocalServiceClient)
+    requires client.ValidState()
+    modifies client.Modifies
+    ensures client.ValidState()
+  {
+    var ddbClient :- expect Dynamodb.DynamoDBClient();
+    var resFailure := client.CallDDBScan(SimpleCallingAWSSDKFromLocalService.Types.CallDDBScanInput(ddbClient := ddbClient, tableArn := TABLE_ARN_FAILURE_CASE));
+
+    expect resFailure.Failure?;
+  }
+
+  method{:test} CallKMSEncrypt(){
+    var client :- expect SimpleCallingAWSSDKFromLocalService.SimpleCallingAWSSDKFromLocalService();
+    TestCallKMSEncrypt_Success(client);
+    TestCallKMSEncrypt_Failure(client);
+  }
+
+  method TestCallKMSEncrypt_Success(client: ISimpleCallingAWSSDKFromLocalServiceClient)
+    requires client.ValidState()
+    modifies client.Modifies
+    ensures client.ValidState()
+  {
+    var kmsClient :- expect Kms.KMSClient();
+    var resSuccess := client.CallKMSEncrypt(SimpleCallingAWSSDKFromLocalService.Types.CallKMSEncryptInput(kmsClient := kmsClient, keyId := KEY_ID_SUCCESS_CASE, plaintext := PLAIN_TEXT));
+    
+    expect resSuccess.Success?;
+    expect resSuccess.value.encryptOutput == KEY_ID_SUCCESS_CASE;
+  }
+
+  method TestCallKMSEncrypt_Failure(client: ISimpleCallingAWSSDKFromLocalServiceClient)
+    requires client.ValidState()
+    modifies client.Modifies
+    ensures client.ValidState()
+  {
+    var kmsClient :- expect Kms.KMSClient();
+    // Test with NonExistent
+    var input_NonExistent := Kms.Types.EncryptRequest(
+      KeyId := NONEXISTENT_KEY_ID,
+      Plaintext := [ 97, 115, 100, 102 ],
+      EncryptionContext := Wrappers.None,
+      GrantTokens := Wrappers.None,
+      EncryptionAlgorithm := Wrappers.None
+    );
+    var resFailure_NonExistent := client.CallKMSEncrypt(SimpleCallingAWSSDKFromLocalService.Types.CallKMSEncryptInput(kmsClient := kmsClient, keyId := NONEXISTENT_KEY_ID, plaintext := PLAIN_TEXT));
+    
+    expect resFailure_NonExistent.Failure?;
+    expect resFailure_NonExistent.error.ComAmazonawsKms?;
+    expect resFailure_NonExistent.error.ComAmazonawsKms.NotFoundException?;
```

Comment:
Ohh I see it's here. Why wasn't this possible in DDB case?

Reply:
I think my previous comment would address this comment. lmk if it does not.